### PR TITLE
Reporting in T1 Node

### PR DESCRIPTION
**Pull request by fsargent**

### CODE REVIEW

File: integration-test/getReports.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+var expect = require('./chai_config').expect;
+var t1 = require('../index');
+require('dotenv').load();
+
+describe('Get, create and save Entities', function() {
```
Comment (FodT) [inline integration-test/getReports.js]: Still reads entities :)

File: lib/report.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+Report.prototype.get = function (connection, userParams) {
+    if (!connection) {
+        throw new Error('connection object must be provided');
+    }
+    var that = this;
+    if (!userParams.time_window) {
+        throw new Error('A time window parameter is required.')
+    } // Add more known required fields here.
+
```
Comment (FodT) [inline lib/report.js]: Should probably be validating these against a schema like we do with some other uses, if we want to do this at all. This isn't valid either, as you can specify start_date and end_date instead.

File: .gitignore [hunk 3]
```diff
@@ -32,3 +32,6 @@ build/Release
 # Dependency directory
 # https://www.npmjs.org/doc/misc/npm-faq.html#should-i-check-my-node_modules-folder-into-git
 node_modules
+
+# Environment Variables for integration tests
+.env
```
Comment (FodT) [inline .gitignore]: no newline

File: README.md [hunk 4]
```diff
@@ -140,3 +140,11 @@ targetingSegments.include_op = 'OR';
...
 targetingSegments.save(connection).done(console.log('saved'));
 ```
+
+## Tests
+
+`npm test` will run local tests.
+
+To run integration tests, copy `.env.template` to `.env` and fill in the required values.
+
+`npm run integration` will run integration tests with Mocha.
```
Comment (FodT) [inline README.md]: should probably add basic reporting usage to the readme.

Comment (karthik-mediamath) [inline README.md]: can also add the output to show what does the call returns.

File: integration-test/GetCreateAndSaveEntities.js [hunk 5]
```diff
@@ -9,9 +10,11 @@ describe("Get, create and save Entities", function () {
         api_key: process.env.T1SANDBOX_API_KEY,
         client_secret: process.env.T1SANDBOX_SECRET,
         apiBaseUrl: 'https://t1sandbox.mediamath.com',
-        redirect_uri: 'https://blog.mediamath.com/'
+        redirect_uri: 'https://blog.mediamath.com/',
+        advertiser_id: parseInt(process.env.T1SANDBOX_ADVERTISER)
+        // Tests will return 403 if user does not have access to advertiser
     };
-
+    console.log(t1conf);
```
Comment (karthik-mediamath) [inline integration-test/GetCreateAndSaveEntities.js]: `console.log` used for debugging?

Comment (fsargent) [inline integration-test/GetCreateAndSaveEntities.js]: The integration tests don't have any output. Removed the console.log, and fixed the bad t1conf variables.

File: .gitignore (hunk 3 above)
Comment (fsargent) [inline .gitignore]: Done

**Pull request merged**